aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/DeploymentSteps.java

```diff
@@ -209,6 +216,45 @@ private String getCliUpdateConfigArgs(String componentName, Map<String, Object>
         return mapper.writeValueAsString(configurationUpdate);
     }
 
+    private void executeCommandWithConfig(int retryCount, List<String> commandArgs)
+            throws InterruptedException {
+        try {
+            CommandInput command = getDeploymentCommand(commandArgs);
+
+            String response = new String(device.execute(command), StandardCharsets.UTF_8);
+            LOGGER.debug("The response from executing gg-cli command is {}", response);
+            String[] responseArray = response.split(":");
+            String deploymentId = responseArray[responseArray.length - 1];
+            LOGGER.info("The local deployment response is " + deploymentId);
+            scenarioContext.put(LOCAL_DEPLOYMENT_ID, deploymentId);
+        } catch (Exception e) {
+            if (retryCount > 3) {
+                throw e;
+            }
+
+            waits.until(5, "SECONDS");
+            LOGGER.warn("the deployment request threw an exception, retried {} times...", retryCount);
+            this.executeCommand(retryCount + 1, commandArgs);
+        }
+    }
+
+    private CommandInput getDeploymentCommand(List<String> commandArgs) {
```

Comment:
This is not a deployment command. L244 and the entire method has to be under `execute`.